Description: Files present in upstream branch but missing from upstream tarball
 This allows us to maintain the packaging as a branch of upstream more
 easily.  The build system has been corrected upstream, so this patch can be
 dropped after the next upstream release.
Last-Update: 2012-09-04

Index: b/grub-core/tests/boot/linux.init-ppc.S
===================================================================
--- /dev/null
+++ b/grub-core/tests/boot/linux.init-ppc.S
@@ -0,0 +1,61 @@
+/*
+ *  GRUB  --  GRand Unified Bootloader
+ *  Copyright (C) 2012  Free Software Foundation, Inc.
+ *
+ *  GRUB is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  GRUB is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with GRUB.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#define SYSCALL_WRITE 4
+#define SYSCALL_RESET 88
+#define SYSCALL_EXIT 1
+
+#define STDOUT 1
+#define SHUTDOWN_MAGIC1 0xfee1dead
+#define SHUTDOWN_MAGIC2 0x28121969
+#define SHUTDOWN_MAGIC3 0x4321fedc
+
+	.text
+	.global start, _start, __start
+__start:	
+_start:
+start:
+	/* write.  */
+	li %r0, SYSCALL_WRITE
+	li %r3, STDOUT
+	lis %r4, message@h
+	ori %r4, %r4, message@l
+	lis %r5, messageend@h
+	ori %r5, %r5, messageend@l
+	sub %r5, %r5, %r4
+	sc
+
+	/* shutdown.  */
+	li %r0, SYSCALL_RESET
+	lis %r3, SHUTDOWN_MAGIC1@h
+	ori %r3, %r3, SHUTDOWN_MAGIC1@l
+	lis %r4, SHUTDOWN_MAGIC2@h
+	ori %r4, %r4, SHUTDOWN_MAGIC2@l
+	lis %r5, SHUTDOWN_MAGIC3@h
+	ori %r5, %r5, SHUTDOWN_MAGIC3@l
+	sc
+
+	/* exit(1). Shouldn't be reached.  */
+	li %r0, SYSCALL_EXIT
+	li %r3, 1
+	sc
+
+	.data
+message:
+	.ascii "Boot Test Passed Successfully\n" SUCCESSFUL_BOOT_STRING "\n"
+messageend:
Index: b/grub-core/tests/boot/linux.init-mips.S
===================================================================
--- /dev/null
+++ b/grub-core/tests/boot/linux.init-mips.S
@@ -0,0 +1,61 @@
+/*
+ *  GRUB  --  GRand Unified Bootloader
+ *  Copyright (C) 2010  Free Software Foundation, Inc.
+ *
+ *  GRUB is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  GRUB is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with GRUB.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#define SYSCALL_WRITE 4004
+#define SYSCALL_RESET 4088
+#define SYSCALL_EXIT 4001
+
+#define STDOUT 1
+#define SHUTDOWN_MAGIC1 0xfee1dead
+#define SHUTDOWN_MAGIC2 0x28121969
+#ifdef REBOOT
+#define SHUTDOWN_MAGIC3 0x01234567
+#else
+#define SHUTDOWN_MAGIC3 0x4321fedc
+#endif
+	.text
+	.global start, _start, __start
+__start:	
+_start:
+start:
+	/* write.  */
+	li $v0, SYSCALL_WRITE
+	li $a0, STDOUT
+	lui $a1, %hi(message)
+	addiu $a1, %lo(message)
+	lui $a2, %hi(messageend)
+	addiu $a2, %lo(messageend)
+	subu $a2, $a2, $a1
+	syscall
+
+	/* shutdown.  */
+	li $v0, SYSCALL_RESET
+	li $a0, SHUTDOWN_MAGIC1
+	li $a1, SHUTDOWN_MAGIC2
+	li $a2, SHUTDOWN_MAGIC3
+	syscall
+
+	/* exit(1). Shouldn't be reached.  */
+	li $v0, SYSCALL_EXIT
+	li $a0, 1
+	syscall
+
+	.data
+message:
+	.ascii "Boot Test Passed Successfully\n" SUCCESSFUL_BOOT_STRING "\n"
+messageend:
Index: b/grub-core/tests/boot/linux-ppc.cfg
===================================================================
--- /dev/null
+++ b/grub-core/tests/boot/linux-ppc.cfg
@@ -0,0 +1,5 @@
+linux /linux console=ttyPZ1 root=/dev/ram0
+initrd /initrd
+boot
+# Shouln't happen
+halt
Index: b/linguas.sh
===================================================================
--- /dev/null
+++ b/linguas.sh
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+rsync -Lrtvz --exclude=ko.po  translationproject.org::tp/latest/grub/ po
+
+autogenerated="en@quot en@hebrew de@hebrew en@cyrillic en@greek en@arabic en@piglatin de_CH"
+
+
+for x in $autogenerated; do
+    rm "po/$x.po";
+done
+
+
+(
+    (
+	cd po && ls *.po| cut -d. -f1
+	for x in $autogenerated; do
+	    echo "$x";
+	done
+    ) | sort | uniq | xargs
+) >po/LINGUAS
